### PR TITLE
Add the input variable `zone_name`

### LINKED ISSUE #11: Provide extra variable for the route53_zone data source
### Problem
I want to create this certificate:
```
domain_name                  = "site.example.org"
subject_alternative_names = ["www.site.example.org"]
```

My existing Route53 hosted zone is `example.org.`

The module fails because it looks for the `site.example.org` zone, which does not exist. I want to keep my `example.org` zone and I don't want to create a wildcard certificate.

### Solution
* add extra variable `zone_name`
* use variable `zone_name` as name for the route53_zone data source if it exists:
```hcl
variable "zone_name" {
  type        = "string"
  default     = ""
  description = "The Hosted Zone name of the desired Hosted Zone"
}

data "aws_route53_zone" "default" {
  count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
  name         = "${local.zone_name}"
  private_zone = false
}

locals {
  zone_name = "${var.zone_name == "" ? var.domain_name. : var.zone_name}"
}
```

I am happy to write a PR if you agree with the proposed solution.

### PR DESCRIPTION
**Pull request by ldormoy**
## what
* It allows choosing a zone whose name differs from the cert domain name

## why
* closes #11

### CODE REVIEW

**Comment by aknysh** [comment]
thanks @ldormoy, looks good
rebuild README

File: main.tf [hunk 1]
```diff
@@ -11,12 +11,13 @@ resource "aws_acm_certificate" "default" {
...
 data "aws_route53_zone" "default" {
   count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
-  name         = "${var.domain_name}."
+  name         = "${local.zone_name}"
   private_zone = false
 }
 
 locals {
   domain_validation_options = "${aws_acm_certificate.default.domain_validation_options[0]}"
+  zone_name                 = "${var.zone_name == "" ? var.domain_name. : var.zone_name}"
```
Comment (osterman) [inline main.tf]:
```suggestion
  zone_name                 = "${var.zone_name == "" ? var.domain_name : var.zone_name}"
```

File: main.tf [hunk 2]
```diff
@@ -11,12 +11,13 @@ resource "aws_acm_certificate" "default" {
 
 data "aws_route53_zone" "default" {
   count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
-  name         = "${var.domain_name}."
+  name         = "${local.zone_name}"
```
Comment (osterman) [inline main.tf]: Hrm... maybe the ending `.` is optional these days?

Comment (osterman) [reply main.tf]: https://serverfault.com/a/803037

Comment (ldormoy) [reply main.tf]: Oh I did not see your serverfault link. In this case let's remove the dot.

Comment (aknysh) [reply main.tf]: for backwards compatibility, `var.domain_name` should not include a dot, it's added in the code.
for consistency, I think it should not be included in `var.zone_name` either.

```
zone_name                 = "${var.zone_name == "" ? var.domain_name. : var.zone_name.}" 
```

Comment (ldormoy) [reply main.tf]: Okay then I'll just add the dot like this, as @osterman suggested:
```
data "aws_route53_zone" "default" {
  count        = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? 1 : 0}"
  name         = "${local.zone_name}."
  private_zone = false
}

```

**Comment by osterman** [comment]
Release 0.3.0 with these changes

Thanks!!

**Pull request merged**